### DISCUSSION TITLE
Adding Classes in pretrained models without distrubing other classes

### DISCUSSION THREAD

**Discussion by Taiyo0004** [ultralytics/ultralytics#23405] [General]
I want to ask how can we add classes to a pretrained model like yolov10 pretrained model so i could get the accuracy of those pretrained model and also add some of mine, is it even possible or not.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Taiyo0004, thanks for opening this discussion on adding classes to a pretrained model 🚀 This is an automated response to help you get started, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (like adding new classes while keeping existing ones), please share as much detail as you can so we can point you in the right direction, including:
-Your current `ultralytics` version
-Which pretrained weights you’re starting from (e.g., YOLOv10 variant)
-Your dataset YAML (classes/names) and a few example images (or a clear description)
-Your training command/config and relevant logs

Also please verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) ✅

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
For standard closed-vocabulary detectors like `yolov10*.pt`, you can’t “append” classes while keeping the original head/weights unchanged—changing `nc` replaces the detection head, so preserving old-class accuracy requires fine-tuning on a dataset that includes both your new classes and representative data for the old ones (otherwise the old classes will typically degrade).

```bash
yolo train model=yolov10s.pt data=your_data.yaml imgsz=640 epochs=50
```

If your goal is adding new class names **without retraining**, use an open-vocabulary model like [YOLO-World](https://docs.ultralytics.com/models/yolo-world/) (prompted via [`set_classes()`](https://docs.ultralytics.com/reference/models/yolo/model/))—accuracy will be scenario-dependent, but it’s the intended “add classes at runtime” path.
